Source/WebKit2/Platform/IPC/Connection.cpp

```diff
@@ -873,6 +873,7 @@ void Connection::didFailToSendSyncMessage()
     if (!m_shouldExitOnSyncMessageSendFailure)
         return;
 
+    m_client.didClose(*this);
```

Comment:
This could be called with m_isValid = false, I don't think we are expected to call the client when connection has already been invalidated. I guess we should set m_isValid to false here and probably we want to protect this, since the client might destroy the connection on close.